pallets/liquidity-pools-gateway/routers/src/tests.rs

```diff
@@ -28,739 +28,739 @@ lazy_static! {
-
-	use super::*;
-
-	mod util {
-		use super::*;
-
-		pub struct EVMRouterTestData {
-			pub test_contract_address: H160,
-			pub test_contract_code: Vec<u8>,
-			pub test_contract_hash: H256,
-			pub evm_domain: EVMDomain,
-			pub sender: AccountId32,
-			pub sender_h160: H160,
-			pub derived_sender: AccountId32,
-			pub msg: Vec<u8>,
-		}
-
-		pub fn get_test_data() -> EVMRouterTestData {
-			let test_contract_address = H160::from_low_u64_be(1);
-			let test_contract_code = [0; 32].to_vec();
-			let test_contract_hash = BlakeTwo256::hash_of(&test_contract_code);
-
-			let evm_domain = EVMDomain {
-				target_contract_address: test_contract_address,
-				target_contract_hash: test_contract_hash,
-				fee_values: FeeValues {
-					value: U256::from(10),
-					gas_limit: U256::from(10),
-					gas_price: U256::from(10),
-				},
-			};
-
-			let sender: AccountId32 = [0; 32].into();
-			let sender_h160: H160 =
-				H160::from_slice(&<AccountId32 as AsRef<[u8; 32]>>::as_ref(&sender)[0..20]);
-			let derived_sender = IdentityAddressMapping::into_account_id(sender_h160);
-
-			let msg = vec![0, 1, 2];
-
-			EVMRouterTestData {
-				test_contract_address,
-				test_contract_code,
-				test_contract_hash,
-				evm_domain,
-				sender,
-				sender_h160,
-				derived_sender,
-				msg,
-			}
-		}
-	}
-
-	mod init {
-		use super::*;
-
-		#[test]
-		fn success() {
-			new_test_ext().execute_with(|| {
-				let test_data = get_test_data();
-
-				pallet_evm::AccountCodes::<Runtime>::insert(
-					test_data.test_contract_address,
-					test_data.test_contract_code,
-				);
-
-				let router = EVMRouter::<Runtime> {
-					evm_domain: test_data.evm_domain,
-					_marker: Default::default(),
-				};
-
-				assert_ok!(router.do_init());
-			});
-		}
-
-		#[test]
-		fn failure() {
-			new_test_ext().execute_with(|| {
-				let test_data = get_test_data();
-
-				let router = EVMRouter::<Runtime> {
-					evm_domain: test_data.evm_domain,
-					_marker: Default::default(),
-				};
-
-				assert_noop!(
+    use util::*;
+
+    use super::*;
+
+    mod util {
+        use super::*;
+
+        pub struct EVMRouterTestData {
+            pub test_contract_address: H160,
+            pub test_contract_code: Vec<u8>,
+            pub test_contract_hash: H256,
+            pub evm_domain: EVMDomain,
+            pub sender: AccountId32,
+            pub sender_h160: H160,
+            pub derived_sender: AccountId32,
+            pub msg: Vec<u8>,
+        }
+
+        pub fn get_test_data() -> EVMRouterTestData {
+            let test_contract_address = H160::from_low_u64_be(1);
+            let test_contract_code = [0; 32].to_vec();
+            let test_contract_hash = BlakeTwo256::hash_of(&test_contract_code);
+
+            let evm_domain = EVMDomain {
+                target_contract_address: test_contract_address,
+                target_contract_hash: test_contract_hash,
+                fee_values: FeeValues {
+                    value: U256::from(0),
```

Comment:
@mustermeiszer This value here is what caused the failure in the original PR. The EVM logic would try to transfer this amount from the sender to the derived contract address.